src/vfs/createVirtualFileSystem.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  FileStat,
+  FileSystemTree,
+  VirtualFileSystem,
+} from "./virtualFileSystem";
+import path from "path";
+
+/**
+ * Creates a virtual file system for managing files only (no directories).
+ */
+
+export function createVirtualFileSystem(
+  root: string,
+  fileSystemTree: FileSystemTree = {},
+  readonly: boolean = true,
+): VirtualFileSystem {
+  let normalizedRoot = path.normalize(root);
+  if (!normalizedRoot.endsWith(path.sep)) {
+    normalizedRoot += path.sep;
+  }
+
+  const memoryFS = fileSystemTree;
+
+  return {
+    root: normalizedRoot,
+    type: "memory",
+
+    exists(filePath: string): boolean {
+      const resolvedPath = this.resolve(filePath);
+      return resolvedPath in memoryFS;
+    },
+
+    resolve(...filePath: string[]): string {
+      return path.normalize(path.resolve(normalizedRoot, ...filePath));
+    },
+
+    readFile(filePath: string): Buffer {
+      const resolvedPath = this.resolve(filePath);
+      const file = memoryFS[resolvedPath];
+
+      if (!file || file.type !== "file") {
+        throw new Error(
+          `File '${resolvedPath}' does not exist or is not a file`,
+        );
+      }
+
+      const content = file.content ?? ""; // Default to an empty string if content is undefined
```

Comment:
In what cases the content could be `undefined`? Does it indicate an error?

Reply:
There will no longer be any undefined values, as content needs to be provided.